cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManagerFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol;
+
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+
+/**
+ * RepositoryManagerFactory interface which is used by Guice during runtime to create a RepositoryManager.
+ */
+public interface RepositoryManagerFactory {
+  RepositoryManager create(NamespaceId namespace, RepositoryConfig repoConfig);
```

Comment:
Why the caller need to pass RepositoryConfig? It seems unnecessary and source of errors.

Reply:
Currently the RepositoryManager is depending on the RepositoryConfig and the config is only accessible from `SourceControlManagementService`, we could refactor this by adding the `RepositoryStore` layer and read the config from `RepositoryManager`

Reply:
Bring @arjan-bal for visibility

Reply:
@tivv Not willing to add the repo store change in this PR cause it adds more distraction. I've pushed the commit for a separate branch, which does not take the `RepositoryConfig` anymore: https://github.com/cdapio/cdap/commit/d331dd24daf2de6a7288f26557748c1587141261. Will raise the change once this PR merges